### PR TITLE
Build runner

### PR DESCRIPTION
**Pull request by sagesmith-wf**
#### Changes:
- Added toStrings
- Changed logger level
- Added runner
- implemented options
- Added a small racetrack
- Fixed error with logging

Please Review: @Kuroshii @JWhitney406

### CODE REVIEW

File: src/main/java/com.ai/Collision.java [hunk 1]
```diff
@@ -119,10 +124,15 @@ public State getNextState(Racetrack racetrack, Position position, Velocity veloc
...
+        if (endPosition == null) {
+            return null;
+        }
 
         return new State(racetrack.startingLine().iterator().next(), new Velocity(0, 0));
     }
+
+    @Override
+    public String toString() {
+        return "CollisionModel(start)";
```
Comment (Kuroshii) [inline src/main/java/com.ai/Collision.java]: `restart` instead of `start`?

File: src/main/java/com.ai/Racetrack.java [hunk 2]
```diff
@@ -13,6 +12,7 @@
 public class Racetrack {
     private final boolean[][] isSafe;
     private final int width, height;
+    private static String name = "No name specified";
```
Comment (Kuroshii) [inline src/main/java/com.ai/Racetrack.java]: Why is this static? Seems like a name for a racetrack should be an instance variable.

Comment (sagesmith-wf) [reply src/main/java/com.ai/Racetrack.java]: From file is static and the constructer doesn't require a filename so I had to make it static

Comment (Kuroshii) [reply src/main/java/com.ai/Racetrack.java]: But then won't `toString()` work incorrectly e.g.
```
Racetrack trackA = fromFile("test1.txt");
System.out.println(trackA);
Racetrack trackB = fromFile("test2.txt");
System.out.println(trackA);
```
would output:
```
test1
test2

**Comment by sagesmith-wf** [comment]
Comments Addressed.

File: src/main/java/com.ai/Main.java [hunk 3]
```diff
@@ -1,36 +1,300 @@
...
+     * @param options the options to check for racetracks in
+     *
+     * @return the specified racetracks
+     * @throws IOException if the racetracks cannot be found
+     */
+    private static List<Racetrack> getRaceTracks(OptionSet options) throws IOException {
+        List<Racetrack> racetracks = new ArrayList<>();
+
+        Map<String, Racetrack> racetrackRegistry = new HashMap<>();
+        logger.debug("Adding only the l_track to the registry...");
```
Comment (Kuroshii) [inline src/main/java/com.ai/Main.java]: Problem solved; probably don't need these debug lines anymore?

Comment (sagesmith-wf) [reply src/main/java/com.ai/Main.java]: Comment addressed

**Pull request merged**